src/App.test.js

```diff
@@ -1,9 +1,87 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './App';
-
-it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+const puppeteer = require('puppeteer');
+const faker = require('faker');
+
+const person = {
+  name: faker.name.firstName() + ' ' + faker.name.lastName(),
+  email: "c@lvin.com",
+  phone: faker.phone.phoneNumber(),
+  message: faker.random.words(),
+  password: "password"
+};
+
+
+
+let browser;
+let page;
+const width = 1920;
+const height = 1080;
+
+beforeAll(async () => {
+  browser = await puppeteer.launch({
+    headless: false,
+    slowMo: 80,
+    args: [`--window-size=${width},${height}`]
+  });
+  page = await browser.newPage();
+  await page.setViewport({ width, height });
 });
+afterAll(() => {
+  browser.close();
+});
+
+
+
+const appUrlBase = 'http://localhost:3000'
+const routes = {
+  public: {
+    login: `${appUrlBase}/login`,
+    noMatch: `${appUrlBase}/PageNotFound`,
+  },
+  private: {
+    dashboard: `${appUrlBase}/`,
+    registration: `${appUrlBase}/registration`,
+  },
+};
+
+describe('Signing In', () => {
+  test('sign-in title loads correctly', async () => {
+
+    await page.goto(routes.public.login);
+    await page.waitForSelector('.header');
+
+    const html = await page.$eval('.header', e => e.innerHTML);
+    expect(html).toBe('sign in');
+
+
+  }, 16000);
+  test('login form works correctly', async () => {
+    // take a png screenshot and saves it inside root directory
+    await page.screenshot({ path: "loginScreen.png" })
```

Comment:
Save screenshots to a new folder (e.g. `debug_screenshots` or similar) to avoid cluttering the root folder.